server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheRequestProcessor.java

```diff
@@ -253,15 +258,17 @@ private void putInternal(HotRodHeader header, AdvancedCache<byte[], byte[]> cach
                .putAsync(key, value, metadata)
                .thenApply(CompletableFutures.toNullFunction());
       }
-      cs.whenComplete((ce, throwable) -> handlePut(header, ce, throwable))
-            .whenComplete(span);
+      cs.whenComplete((ce, throwable) -> handlePut(header, ce, throwable, span));
    }
 
-   private void handlePut(HotRodHeader header, CacheEntry<byte[], byte[]> ce, Throwable throwable) {
+   private void handlePut(HotRodHeader header, CacheEntry<byte[], byte[]> ce, Throwable throwable,
+                          InfinispanSpan<CacheEntry<?, ?>> span) {
       if (throwable != null) {
          writeException(header, throwable);
+         span.accept(null, throwable);
```

Comment:
```suggestion
         span.recordException(throwable);
         span.complete():
```

Reply:
It should be doing this in the `accept` method

```java
   @Override
   default void accept(T ignored, Throwable throwable) {
      if (throwable != null) {
         recordException(throwable);
      }
      complete();
   }
```